classes/local/users.php

```diff
@@ -135,4 +135,23 @@ public function get_unissued_users($users, $accredibleinstanceid = null) {
 
         return $unissuedusers;
     }
+
+    /**
+     * Get user grades from grade item.
+     * @param stdObject $data data from the submission of mod_form.
+     * @return array[stdClass] $gradeattributes
+     */
+    public function get_user_grades($data) {
+        global $DB;
+
+        if (isset($data->includegradeattribute) && isset($data->gradeattributegradeitemid) && isset($data->gradeattributekeyname)) {
+            $users = array_keys($data->users);
+            $assigment = $DB->get_record('grade_items', array('id' => $data->gradeattributegradeitemid), '*', MUST_EXIST);
```

Comment:
I think this is not `assignment` but `gradeitem` as it can be a quiz, a forum, etc.